### PR TITLE
Issue #134: Removing boot2docker ISO creation

### PR DESCRIPTION
**Pull request by LalatenduMohanty**
Now the the default ISO would be from github.com/minishift/minishift-b2d-iso

Signed-off-by: Lalatendu Mohanty <lmohanty@redhat.com>

### CODE REVIEW

**Comment by LalatenduMohanty** [comment]
@hferentschik Actually I wanted to cover part of https://github.com/minishift/minishift/issues/278 in this PR. But I am still do not have solution for that and also due to lack of time (as I am going PTO soon) trying to just fix #134.

File: pkg/minikube/constants/constants.go [hunk 1]
```diff
@@ -44,7 +43,7 @@ const (
 	DefaultDiskSize = "20g"
 )
 
-var DefaultIsoUrl = "https://github.com/minishift/minishift/releases/download/v" + version.GetVersion() + "/boot2docker.iso"
+var DefaultIsoUrl = "https://github.com/minishift/minishift-b2d-iso/releases/download/v1.0.0/minishift-b2d.iso"
```
Comment (praveenkumar) [inline pkg/minikube/constants/constants.go]: Does hardcoded verison make sense can't we add a go function to get latest release url? I have prototype in python which I can put in go also : https://gist.github.com/praveenkumar/f5e26a20f2b6a6640b86bc2934371bf9

@hferentschik wdyt?

Comment (LalatenduMohanty) [inline pkg/minikube/constants/constants.go]: +1 to the idea of automatically fetching the latest release. I am only concerned with the call to api.github.com. Users are already facing some issues with it with respect to number of calls being made.

Comment (hferentschik) [inline pkg/minikube/constants/constants.go]: -1 for hard coded the version like this. If we hard code it (which I am ok doing initially), we need to do it in the Makefile (inlining the main Minishift version and the OpenShift version).

If we want to GitHub, we want to define a base version say 1.x and then use semantic versioning to get the latest version which is compatible. Not just the latest version. The functionality is already there to do this. So on sets the base version 1.x (eg in the Makefile). Then you go out to GitHub and get the latest version which is compatible in a semantic versioning sense. This is what I had in mind.

There is no risk to hit the Git Hub API rate limit here. We only do this when the ISO is not cached. There is no automatic update right now.

File: Makefile [hunk 2]
```diff
@@ -14,6 +14,8 @@
 
 VERSION ?= $(shell cat VERSION)
 OPENSHIFT_VERSION ?= $(shell cat OPENSHIFT_VERSION)
+OPENSHIFT_VERSION ?= $(shell cat OPENSHIFT_VERSION)
+ISO_VERSION ?= $(shell cat ISO_VERSION)
```
Comment (hferentschik) [inline Makefile]: Can we inline this version files? I find we have a bit of a proliferation of version files here.

**Comment by LalatenduMohanty** [comment]
@hferentschik I did not create the GetIsoSemverVersion because I do not see the use of the function in the current code. We can create one when we need to use it. 
I am done with my testing and it is ready for review.

**Comment by hferentschik** [comment]
>  I did not create the GetIsoSemverVersion because I do not see the use of the function in the current code

Sure. I guess for now we can go with a single hard coded version. Let's see whether can refine this for this sprint or whether we pick this up later. 

> I am done with my testing and it is ready for review.

ok

**Comment by hferentschik** [comment]
Created issue #292 to follow up on this.

Also added a commit to add some more logging and a commit to get completely rid of the _deploy_ directory.

File: pkg/version/version.go [hunk 3]
```diff
@@ -47,3 +48,7 @@ func GetOpenShiftVersion() string {
 func GetOpenShiftSemverVersion() (semver.Version, error) {
 	return semver.Make(strings.TrimPrefix(GetOpenShiftVersion(), VersionPrefix))
```
Comment (hferentschik) [inline pkg/version/version.go]: @LalatenduMohanty here is your code which makes use of semantic versioning. You added your code right below ;-) This is using https://github.com/blang/semver which allows you to compare versions and determine whether they are compatible in a semantic versioning sense.

Comment (LalatenduMohanty) [inline pkg/version/version.go]: @hferentschik I have seen this but I guess I missed https://github.com/minishift/minishift/blob/master/pkg/minikube/update/update.go#L57

**Pull request merged**